consensus/scp/src/slot.rs

```diff
@@ -1214,8 +1222,16 @@ impl<V: Value, ValidationError: Display> Slot<V, ValidationError> {
         // but the node has accepted a ballot prepared... , then "ballot.value" is
         // the value of the highest such accepted prepared ballot."
         if let Some(p) = self.ballots_accepted_prepared().into_iter().max() {
+            if let Some(p_old) = &self.P {
+                if p_old > p {
+                    return Some(p_old.X);
+                }
+            }
             return Some(p.X);
         }
+        else if let Some(p_old) = &self.P {
+            return Some(p_old.X);
+        }
 
         // Otherwise, values are unchanged.
         if !self.B.is_zero() {
```

Comment:
I do not believe this conditional will ever be true. B can only be non-zero if: Z is set, H is set, or P is set. Whichever way it gets set, the earlier conditionals in this function will catch it.